tests/baselines/reference/objectSpread.types

```diff
@@ -407,4 +407,11 @@ let shortCutted: { a: number, b: string } = { ...o, a }
 >o : { a: number; b: string; }
 >a : number
 
+// non primitive
+let spreadNonPrimitve = { ...<object>{}}
+>spreadNonPrimitve : { constructor: Function; toString(): string; toLocaleString(): string; valueOf(): Object; hasOwnProperty(v: string): boolean; isPrototypeOf(v: Object): boolean; propertyIsEnumerable(v: string): boolean; }

```

Comment:
`spreadNonPrimitive` should be `{}`. It looks like `getApparentType` gets called somewhere, probably in `getSpreadType`'s call to `getPropertiesOfType(right)`. I think you need a special case in `getSpreadType` that returns `emptyObjectType`.